frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -647,15 +647,60 @@ func dispatchEnv(d *dispatchState, c *instructions.EnvCommand) error {
 }
 
 func dispatchRun(d *dispatchState, c *instructions.RunCommand, proxy *llb.ProxyEnv, sources []*dispatchState, dopt dispatchOpt) error {
+	var opt []llb.RunOption
+
 	var args []string = c.CmdLine
+	if len(c.Files) > 0 {
+		if len(args) != 1 {
+			panic("Parsing has produced an invalid RunCommand")
+		}
+
+		if heredoc := parser.MustParseHeredoc(args[0]); heredoc != nil {
+			if strings.HasPrefix(c.Files[0].Data, "#!") {
+				// This is a single heredoc with a shebang, so create a file
+				// and run it
+				// NOTE: choosing not to expand doesn't really make sense here,
+				// so we silently ignore that option if it was provided
+				sourcePath := "/"
+				destPath := "/dev/pipes/"
```

Comment:
Is there any "standard" for this path?

Does this work on Windows?

Reply:
I think we should probably avoid trying to handle shebangs in windows, mostly because they're unix-y things?

I've added a check to only go down this path if we're not on windows, so hopefully that's *enough* of a fix. In regards to heredocs in general, it all depends on what shell is used really and what level of support they have for them (although the fake heredoc support only requires a shell to be able to read multiple commands separated by newlines).

(also, there's not *really* a standard for that path, so we could plausibly change it if we had a better name)